File: internal/quality/golden/java-spring-mini/src/com/example/demo/controller/UserController.java
package com.example.demo.controller;

import com.example.demo.model.User;
import com.example.demo.service.UserService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/users")
public class UserController {
    private final UserService service;

    public UserController(UserService service) {
        this.service = service;
    }

    @GetMapping
    public List<User> listUsers() {
        return service.listAll();
    }

    @GetMapping("/{email}")
    public User getByEmail(@PathVariable String email) {
        return service.findByEmail(email).orElse(null);
    }

    @PostMapping
    public User createUser(@RequestBody User payload) {
        return service.create(payload.getEmail(), payload.getName());
    }
}

File: internal/mcp/testdata/branches_4446/UserController.kt
package com.example.api

import org.springframework.beans.factory.annotation.Value
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import org.slf4j.LoggerFactory

@RestController
class UserController(private val repo: UserRepository) {

    private val log = LoggerFactory.getLogger(UserController::class.java)

    @Value("\${signup.maxName:50}")
    private var maxName: Int = 50

    // create — Spring controller method (Kotlin) with an env-gate
    // (System.getenv), an early-return guard returning
    // ResponseEntity.status(HttpStatus.BAD_REQUEST), a 409 guard inside a try,
    // and a catch that logs then returns a 500 ResponseEntity.
    @PostMapping("/users")
    fun create(@RequestBody dto: UserDto): ResponseEntity<Any> {
        if (System.getenv("SIGNUP_ENABLED") == null) {
            return ResponseEntity.status(503).build()
        }

        if (dto.email == null) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("email required")
        }

        try {
            if (repo.existsByEmail(dto.email)) {
                return ResponseEntity.status(HttpStatus.CONFLICT).body("email in use")
            }
            val saved = repo.save(dto.toEntity())
            return ResponseEntity.status(HttpStatus.CREATED).body(saved)
        } catch (e: Exception) {
            log.error("create failed", e)
            return ResponseEntity.status(500).body("internal error")
        }
    }
}
